### PR TITLE
{Container monitoring} -Changes for container monitoring to support USNat and USSec

### PR DESCRIPTION
**Pull request by rashmichandrashekar**
Description -
Changes for container monitoring for USNat and USSec.

### CODE REVIEW

**Comment by yonzhan** [comment]
acs

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -2595,6 +2595,28 @@ def _ensure_default_log_analytics_workspace_for_monitoring(cmd, subscription_id,
         "usgovvirginia": "usgovvirginia"
     }
 
+    # mapping for JEDI - USNat
+    AzureUSNatLocationToOmsRegionCodeMap = {
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:2599]: Please use pythonic style.

**Comment by yonzhan** [comment]
add to S172

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls look at this PR, how to cooperate it into cloud endpoint discovery? thanks.

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 2]
```diff
@@ -2603,20 +2625,40 @@ def _ensure_default_log_analytics_workspace_for_monitoring(cmd, subscription_id,
 
     # sanity check that locations and clouds match.
     if ((cloud_name.lower() == 'azurecloud' and AzureChinaRegionToOmsRegionMap.get(rg_location, False)) or
-            (cloud_name.lower() == 'azurecloud' and AzureFairfaxRegionToOmsRegionMap.get(rg_location, False))):
+            (cloud_name.lower() == 'azurecloud' and AzureFairfaxRegionToOmsRegionMap.get(rg_location, False)) or
+            (cloud_name.lower() == 'azurecloud' and AzureUSNatRegionToOmsRegionMap.get(rg_location, False)) or
+            (cloud_name.lower() == 'azurecloud' and AzureUSSecRegionToOmsRegionMap.get(rg_location, False))):
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:2630]: Will the following code work as well?
`if (cloud_name.lower() == 'azurecloud' and not AzureCloudRegionToOmsRegionMap.get(rg_location, False):`

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 3]
```diff
@@ -2595,6 +2595,28 @@ def _ensure_default_log_analytics_workspace_for_monitoring(cmd, subscription_id,
...
+    # mapping for JEDI - USSec
+    AzureUSSecLocationToOmsRegionCodeMap = {
+        "usseceast": "USSECEAST",
+        "ussecwest": "USSECWEST"
+    }
+
+    AzureUSSecRegionToOmsRegionMap = {
+        "usseceast": "usseceast",
+        "ussecwest": "ussecwest"
+    }
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:2618]: Is it allowed to expose the settings of JEDI cloud in code? Please double confirm. 

You may consider add `LocationToOmsRegionCodeMap` and `RegionToOmsRegionMap` as endpoints of a cloud. [cloud.py](https://github.com/Azure/azure-cli/blob/f5ecf9e2aca18c3409689bda5e8d892b97942e15/src/azure-cli-core/azure/cli/core/cloud.py#L56).

File: src/azure-cli/azure/cli/command_modules/acs/custom.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rashmichandrashekar) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:2599]: Following up with Moshe on this. I will update this to the once that they confirm is being used across services in LA.

File: src/azure-cli/azure/cli/command_modules/acs/custom.py (hunk 3 above)
Comment (rashmichandrashekar) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:2618]: @fengzhou-msft - This still puts it in the code.
Is there any way i can get this information from another place without having to put it in the open source code?
Are you aware of any other teams which are doing this?

File: src/azure-cli/azure/cli/command_modules/acs/custom.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rashmichandrashekar) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py]: There's no good way of putting this outside of code today. I have updated to use the region in the default logic.

File: src/azure-cli/azure/cli/command_modules/acs/custom.py (hunk 2 above)
Comment (rashmichandrashekar) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py]: code removed

File: src/azure-cli/azure/cli/command_modules/acs/custom.py (hunk 3 above)
Comment (rashmichandrashekar) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py]: Spoke to Fengzhou offline. Right now the API support is stil being built and I have updated the code to work around this problem.

**Comment by rashmichandrashekar** [comment]
@arrownj - Could you please take a look at this PR and approve?

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 4]
```diff
@@ -2627,7 +2627,8 @@ def _ensure_default_log_analytics_workspace_for_monitoring(cmd, subscription_id,
         workspace_region = AzureFairfaxRegionToOmsRegionMap.get(rg_location, "usgovvirginia")
         workspace_region_code = AzureFairfaxLocationToOmsRegionCodeMap.get(workspace_region, "USGV")
     else:
-        logger.error("AKS Monitoring addon not supported in cloud : %s", cloud_name)
+        workspace_region = rg_location
+        workspace_region_code = rg_location.upper()
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:2631]: Could you please help explain more why this change is to support USNat and USSec ?

Comment (rashmichandrashekar) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:2631]: Right now we have a check for cloud that maps region to short code for workspace creation. Since we cannot add the mapping in the open source repo, i am using the approach of using the region name in the workspace which enables the support for the USNat and USSec clouds.
